### PR TITLE
Chaos Mesh Project Security Self-Assessment Security Pals

### PR DESCRIPTION
**Pull request by mathur-14**
Created and added first draft for **Chaos Mesh** Project Security Self-Assessment.

Please feel free to share your feedback on the security self-assessment.

### CODE REVIEW

**Comment by eddie-knight** [comment]
Hi there! I'm just getting started looking at your pull request, and I noticed the DCO check is failing.

You can look at the checks section of the PR (I believe it should always be below the last comment) and look for a red X highlighting the failed check. In this case, you can click `Details` for more information about how to get that check passing.
Screenshot 2023-12-08 at 8 35 18 AM

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
+# Self-assessment
```
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:1]:
```suggestion
# Chaos Mesh Self-assessment
```

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+The Self-assessment is the initial document for projects to begin thinking about the
+security of the project, determining gaps in their security, and preparing any security
+documentation for their users. This document is ideal for projects currently in the
+CNCF **sandbox** as well as projects that are looking to receive a joint assessment and
+currently in CNCF **incubation**.
+
+For a detailed guide with step-by-step discussion and examples, check out the free 
+Express Learning course provided by Linux Foundation Training & Certification: 
+[Security Assessments for Open Source Projects](https://training.linuxfoundation.org/express-learning/security-self-assessments-for-open-source-projects-lfel1005/).
+
```
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:11]:
```suggestion
```

Comment (eddie-knight) [reply assessments/projects/chaos-mesh/self-assessment.md:11]: Please go back to ensure that no template text like this is included here or in the rest of the document

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 1 above)
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md]: I've made the required changes.

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 2 above)
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md]: I've made the required changes. No pre-texts now exists.

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+  - [Appendix](#appendix)
+
+## Metadata
+
+|                   |                                                                                                                                   |
+| ----------------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| Software          | https://github.com/chaos-mesh/chaos-mesh                                                                                          |
+| Security Provider | No, Chaos Mesh offers various types of fault simulation and has an enormous capability to orchestrate fault scenarios.            |
+| Languages         | Go, Typescript, Javascript, Shell                                                                                                 |
+| SBOM              | `https://github.com/chaos-mesh/chaos-mesh/releases/tag/v2.6.2` -> Assets -> File: `chaos-mesh-v2.6.2-sbom.spdx` |
```
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:31]:
```suggestion
| SBOM              | Chaos Mesh automatically generates SBOMs on release: https://github.com/chaos-mesh/chaos-mesh/releases/latest |
```

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+
+### Non-goals
+- Resource Utilization Control: Chaos Mesh does not aim to restrict or control the amount of resources utilized by Chaos experiments. For instance, it does not intend to prevent a user from causing excessive resource usage that could potentially incur financial costs or overload servers.
+- Limiting Data Storage: The project does not aim to enforce limitations on the amount of data stored or handled during Chaos experiments. It does not prevent users from storing large amounts of data that might impact infrastructure or incur costs.
+- Complete System Control: Chaos Mesh does not seek to exert control over all aspects of the system but rather focuses on Chaos Engineering-specific functionalities. It doesn't aim to govern every potential action within a Kubernetes environment but rather focuses on controlled fault injections for testing system resilience.
+
+## Self-assessment use
+
+This self-assessment is created by the Chaos Mesh team to perform an internal analysis of the project's security.  It is not intended to provide a security audit of Chaos Mesh, or function as an independent assessment or attestation of Chaos Mesh's security health.
```
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:87]: Did the Chaos Mesh team participate in this activity?

Consider naming the authors or contributors in the first paragraph, right below the page header.

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+2. After the vulnerability is confirmed, the team creates a draft Security Advisory on Github that lists the details of the vulnerability.
+3. Invite related personnel to discuss the fix.
+4. Fork the temporary private repository on Github, and collaborate to fix the vulnerability.
+5. After the fixed code is merged into all supported versions, the vulnerability will be publicly posted in the GitHub Advisory Database.
+
+## Action Items
+
+- [ ] Acquire SBOM from the maintainers of Chaos Mesh
+- [ ] Connect and Discuss the code design and vulnerabilities with the maintainers (appointment of 7th December)
+- [ ] Create a pull request for the maintainers of Chaos Mesh
```
Comment (eddie-knight) [inline assessments/projects/chaos-mesh/self-assessment.md:270]: Following discussions from the review team, we are asking that Security Pals please remove any recommendations from the scope of this document. You may optionally communicate recommendations for projects in a more accessible location, such as in a project community channel or GitHub Issue.

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 4 above)
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md:87]: We did connect with the maintainers of chaos mesh and exchanged information about the project and what this document is about. We did ask them to have a look and share their feedback but so far they haven't got back to us on that.

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 5 above)
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md]: Sure, we have removed it from the doc and we'll be moving forward with any recommendations from their side and add them to GitHub issues.

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 4 above)
Comment (mathur-14) [reply assessments/projects/chaos-mesh/self-assessment.md:92]: We have added the authors and contributors/reviewers at the start.

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+
+|                   |                                                                                                                                   |
+| ----------------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| Software          | https://github.com/chaos-mesh/chaos-mesh                                                                                          |
+| Security Provider | No, Chaos Mesh offers various types of fault simulation and has an enormous capability to orchestrate fault scenarios.            |
+| Languages         | Go, Typescript, Javascript, Shell                                                                                                 |
+| SBOM              | Chaos Mesh automatically generates SBOMs as assets with each release: https://github.com/chaos-mesh/chaos-mesh/releases/latest    |
+|                   |                                                                                                                                   |
+
+### Security links
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:39]: This space is also meant to include their vulnerability disclosure, incident response processes, default configurations, and anything else that can supplement security considerations. Please revisit and include them. In the event they aren't available, kindly mark them as not available

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+
+Chaos Mesh is an open source cloud-native Chaos Engineering platform. It offers various types of fault simulation and has an enormous capability to orchestrate fault scenarios. Using Chaos Mesh, you can conveniently simulate various abnormalities that might occur in reality during the development, testing, and production environments and find potential problems in the system. 
+
+### Background
+
+Chaos engineering is a discipline within software development and operations that focuses on proactively testing systems for their resilience to failure or unexpected conditions. The primary goal is to uncover weaknesses or vulnerabilities in a system's design or architecture before they cause significant issues in real-world scenarios.
+
+Chaos engineering involves intentionally introducing controlled and measured disruptions or faults into a system to observe how it responds under stress. By doing so, engineers can identify weaknesses, bottlenecks, or failure points and then work on improving the system's reliability, robustness, and fault tolerance.
+
+### Actors
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:58]: I see chaos dashboard and tools within containers enabling chaos mesh deliver its full functionalities. 
It maybe worthwhile to include them if they align the definition of actors per https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md#actors

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+- User: The individual initiating Chaos experiments and interacting with the Chaos Dashboard to manipulate and observe the experiments.
+- Chaos Controller Manager: Responsible for scheduling and managing Chaos experiments. It includes multiple Controllers handling different Chaos experiment types and workflows.
+- Chaos Daemon: The executive component running in DaemonSet mode. It carries out specific fault injections by interfering with network devices, file systems, and kernels within the target Pod Namespace.
+
+![Chaos Mesh Architecture](chaos-architecture.png)
+
+### Actions
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:65]: It is worthwhile to include if there are any checks performed in each of these actions.

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+User input and observation:
+- User (Actor) Interaction: Initiating operations and interactions via the Chaos Dashboard or YAML configuration file.
+- Chaos Resource Changes: Triggering changes in Chaos resources (e.g., creating or modifying NetworkChaos resources).
+
+Monitor resource changes, schedule Workflow, and carry out Chaos experiments:
+- Chaos Controller Manager (Actor): Accepting events from the Kubernetes API Server.
+- Workflow Scheduling: Scheduling and managing Chaos experiments and workflows.
+- Chaos Experiment Execution: Triggering specific fault injections, such as Pod failures, network issues, CPU/memory race simulations, etc.
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:73]: Is this action coming from the chaos controller manager itself or there is some other component who is involed? Per the architecture, I assume it is the controller manager. Please clarify

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+## Self-assessment use
+
+This self-assessment is created by the Chaos Mesh team to perform an internal analysis of the project's security.  It is not intended to provide a security audit of Chaos Mesh, or function as an independent assessment or attestation of Chaos Mesh's security health.
+
+This document serves to provide Chaos Mesh users with an initial understanding of Chaos Mesh's security, where to find existing security documentation, Chaos Mesh plans for security, and general overview of Chaos Mesh security practices, both for development of Chaos Mesh as well as security of Chaos Mesh.
+
+This document provides the CNCF TAG-Security with an initial understanding of Chaos Mesh to assist in a joint-assessment, necessary for projects under incubation.  Taken together, this document and the joint-assessment serve as a cornerstone for if and when Chaos Mesh seeks graduation and is preparing for a security audit.
+
+<details>
+<summary>Chaos Mesh Lightweight Threat Model</summary>
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:99]: Please move this threat model to a separate file in the same folder, and link it to the appendix section

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+
+## Project compliance
+
+* In the most recent security self assessment of Chaos Mesh, there is a notable absence of documented adherence to recognized security standards, including but not limited to PCI-DSS, COBIT, ISO, GDPR, etc. Ongoing initiatives are directed towards assessing and aligning Chaos Mesh with these standards, emphasizing a commitment to industry best practices in security and privacy. This adherence is discussed under CII Best Practices under [Appendix](#appendix).
+* Although not meeting any particular standards, Chaos Mesh relies on TLS encryption to ensure robustness and security to safeguard against man in the middle attacks.
+
+
+## Secure development practices
+
+* Development Pipeline.
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:224]: Please add reference if this is documented anywhere. 
This looks largely manual efforts. Are there any automated tests or established devops/devsecops pipeline in the project?

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+    - Chaos Mesh Blog: https://chaos-mesh.org/blog
+    - Twitter: https://twitter.com/chaos_mesh
+    - Community Meeting: https://community.cncf.io/chaos-mesh-community/
+    - Development Meeting: https://community.cncf.io/chaos-mesh-community/
+
+* Ecosystem. 
+
+Chaos Mesh is primarily designed for Kubernetes environments. It leverages Kubernetes features and concepts, like Custom Resource Definitions (CRDs), to manage chaos experiments as Kubernetes resources. This tight integration makes it an essential tool for organizations that use Kubernetes for orchestration.
+
+## Security issue resolution
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:257]: Please provide reference of this process in the appendix

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+* The repository has a [security policy](https://github.com/chaos-mesh/chaos-mesh/security/policy) that outlines their ability to provide security updates to the two most recent minor versions. Secondly, their security advisories page exists, and has a no previous security threats sign on it. 
+* That being said a broad understanding from reviewers about chaos mesh's security [is](https://www.googlecloudcommunity.com/gc/Cloud-Product-Articles/Comparing-CNCF-Chaos-Engineering-Tools/ta-p/500693) that it uses some Linux utilities to implement the low-level chaos types. Similarly, it needs to use the Docker API in the host machine. Therefore, the daemon Pods (deployed as DaemonSet) will run as privileged containers, and will mount the ``/var/run/docker.sock`` socket file. The controller manager Pod will require permissions to manage MutatingWebhookConfiguration, besides some other expected role-based access control (RBAC) permissions, if the sidecar injection is enabled.
+
+CII Best Practices (based on: [link](https://www.bestpractices.dev/en/criteria/0))
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:278]: How was it determined they are doing well? 
Do they have a badge available for their project?

**Review by ragashreeshekar — CHANGES_REQUESTED** [review]
Thanks for the PR @garethr and team, appreciate the efforts.
I have completed first pass of review and left several comments on section that needs your attention. Please feel free to reach out here or on slack for any questions and clarifications.

Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 9 above)
Comment (iamgeeyuh) [reply assessments/projects/chaos-mesh/self-assessment.md:73]: Yes the action is coming from the chaos controller manager. I've updated the formatting of the information to make this more apparent

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 13 above)
Comment (AdityaDJoshi) [reply assessments/projects/chaos-mesh/self-assessment.md:278]: We edited the language to clarify that, Chaos mesh does not possess the badge. We analyzed independently using the CII best practices handbook and edited the document that clarifies this.

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 10 above)
Comment (AdityaDJoshi) [reply assessments/projects/chaos-mesh/self-assessment.md:99]: We have created a separate lightweight threat model.md file and attached it as part of Appendix->References as a link

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 6 above)
Comment (AdityaDJoshi) [reply assessments/projects/chaos-mesh/self-assessment.md:39]: Understood, we added links to the Vulnerability disclosure policy, default configurations, versions supported (for security updates) as well as the process that one has to go through for reporting a vulnerability.

File: assessments/projects/chaos-mesh/self-assessment.md (hunk 8 above)
Comment (ua2047) [reply assessments/projects/chaos-mesh/self-assessment.md:68]: Each of the actions that have associated security checks have been updated.

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 14]
```diff
@@ -0,0 +1,218 @@
+# Chaos Mesh Self-assessment
+
+Authors: Yash Mathur (@mathur-14), Aditya Deepak Joshi (@AdityaDJoshi), Ushan Abeysekera (@ua2047), Jia Huang (@iamgeeyuh)
+
+Contributors/Reviewers: Pranava Kumar Vemula (@Rana-KV), Eddie Knight (@eddie-knight)
+
+The Self-assessment is the initial document for Chaos Mesh to begin thinking about the security of the project, determining gaps in its security, and preparing any security documentation for their users.
```
Comment (torinvdb) [inline assessments/projects/chaos-mesh/self-assessment.md:7]:
```suggestion
# Chaos Mesh Self-assessment

This assessment was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process and is currently pending changes from the maintainer team.

Authors: Yash Mathur (@mathur-14), Aditya Deepak Joshi (@AdityaDJoshi), Ushan Abeysekera (@ua2047), Jia Huang (@iamgeeyuh)

Contributors/Reviewers: Pranava Kumar Vemula (@Rana-KV), Eddie Knight (@eddie-knight)

The Self-assessment is the initial document for Chaos Mesh to begin thinking about the security of the project, determining gaps in its security, and preparing any security documentation for their users.
```
@ragashreeshekar

File: assessments/projects/chaos-mesh/self-assessment.md [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+
+### Non-goals
+- Resource Utilization Control: Chaos Mesh does not aim to restrict or control the amount of resources utilized by Chaos experiments. For instance, it does not intend to prevent a user from causing excessive resource usage that could potentially incur financial costs or overload servers.
+- Limiting Data Storage: The project does not aim to enforce limitations on the amount of data stored or handled during Chaos experiments. It does not prevent users from storing large amounts of data that might impact infrastructure or incur costs.
+- Complete System Control: Chaos Mesh does not seek to exert control over all aspects of the system but rather focuses on Chaos Engineering-specific functionalities. It doesn't aim to govern every potential action within a Kubernetes environment but rather focuses on controlled fault injections for testing system resilience.
+
+## Self-assessment use
+
+This self-assessment is created by the Chaos Mesh team to perform an internal analysis of the project's security.  It is not intended to provide a security audit of Chaos Mesh, or function as an independent assessment or attestation of Chaos Mesh's security health.
```
Comment (ragashreeshekar) [inline assessments/projects/chaos-mesh/self-assessment.md:110]:
```suggestion
This self-assessment is not intended to provide a security audit of Chaos Mesh, or function as an independent assessment or attestation of Chaos Mesh's security health.
```

**Comment by ragashreeshekar** [comment]
@mathur-14 Thanks for making updates to the PR. 
Please address the pending suggestions and I think we are good to go.

**Comment by eddie-knight** [comment]
LGTM. We can merge this PR and then I'll reach out to the maintainers for next steps.

**Review by ragashreeshekar — APPROVED** [review]

**Comment by ragashreeshekar** [comment]
@mathur-14 Thanks for the updates. 
A few things to do before we could merge, please fix them:
- lint, spelling, links and lint errors
- update branch

**Comment by JustinCappos** [comment]
We don't have access to update this and merge it.  I'll close.  Please reopen if someone with the ability to edit this responds and wants to help us merge this.

**Pull request closed without merging**